### ISSUE TITLE
API documentation is not accurate

### ISSUE THREAD

**Issue by classek** [svendahlstrand/ica-api#26]
Hello there !
Nice tuturial but its seems like they have closed access. Is there a workaround ??
<img width="705" alt="Skärmavbild 2024-04-06 kl  17 40 24" src="https://github.com/svendahlstrand/ica-api/assets/6276675/370a03e8-4041-441f-b757-620b915bc442">

**Comment by svendahlstrand** [comment]
Yeah, unfortunately ICA seems to have changed something in their API, so the documentation is not accurate at the moment. This was first reported in issue https://github.com/svendahlstrand/ica-api/issues/25.

I'm going to update the README to make this clear. If something figures out what is going on, pull requests are more than welcome.

**Comment by PierreGode** [comment]
yes the have rebuild it and put an gataway infront of it. URL: https://apimgw-pub.ica.se:443

**Comment by jptuomi** [comment]
Aww, have known about the home assistant integration based on this documentation for a while and when I finally got my ass out the wagon they pull the rug 😭 I'll reach out to a friend I think might still work there.. 🤷‍♂️

**Comment by Gr2dd3** [comment]
Any changes regarding api problem?

![apiproblem2](https://github.com/user-attachments/assets/c96e4e2b-28ff-4a57-aa97-77a6a1e993f3)

**Comment by svendahlstrand** [comment]
No, not that I'm aware of, unfortunately.

**Comment by Ninsew** [comment]
I've looked around alot with no success. I've also contacted ICAs technical team, the one previously responsible for the API, and they told me they do not have a API - and that if they did, they wouldn't have it public. We're probably out of luck, unless ICA releases a new product in the near future and that's the reason they shut it down.

**Comment by mellamomax** [comment]
New API:
https://apimgw-pub.ica.se/sverige/digx/shopping-list/v1/api/list/xxxxxxx/row

replace xxx with shopping list ID. Can be found in DevTools when logged in to ICA and adding item to shoppinglist.
The auth bearer is also there - but this seems to update often

**Comment by Ludvigpbf** [comment]
I did the same thing as @mellamomax and it seems the bearer is updated with just minutes apart. 
If you also have an bank account at ICA Banken you can't have a password, you have to login with BankID, which makes it alot more complicated. 

I was still able to get both the list ID and the bearer and make GET and POST reqs to my list from insomnia. 

One thing that might work for a home assistant applikation is if you make a regular list in your FE and when saved you are promted/redirected to ICA's own webapplikation to login, fetches the bearer and then makes a POST req to your list. You would ofcourse have the listID for the list you want to add items to or create a new one with the endpoint:  
[https://apimgw-pub.ica.se/sverige/digx/shopping-list/v1/api/list/](https://apimgw-pub.ica.se/sverige/digx/shopping-list/v1/api/list/)

If you send a POST req with:
```
{
  "name": "List from insomnia",
}
```

You get the response: 
```
{
	"id": listId,
	"name": "List from insomnia",
	"comment": null,
	"sortingStore": 0,
	"rows": [],
	"created": "2025-05-27T09:19:05.2121711Z",
	"createdBy": "Skapad av UserID",
	"updated": "2025-05-27T09:19:05.2121711Z",
	"updatedBy": "Uppdaterad av UserID",
	"shoppingListId": 35720243,
	"color": {
		"rgb": "217, 20, 99",
		"hex": "#D91463",
		"rgb_Quick": "217,20,99"
	},
	"householdId": ID,
	"status": null,
	"secureId": "ID"
}
```

**Comment by mellamomax** [comment]
I have created an integration which talks with ICA, retrieves and creates a sensor in HA. For now theres a manual refresh service to synk with ICA since the access token needs updating every 5th minute.

I´ve also created a way to integrate google keep lists with this, which syncs both ways - meaning you can ask google (by voice) to add something to your keep list which in turn adds that to the ica list.

But, this whole integration is quiet useless unless there´s a way to retrieve the access token automatically...
If someone has the knowledge I would be open for help here :)

**Comment by lazytarget** [comment]
I have successfully reverse engineered their API and authentication flow. I have not had the time to polish it or document it. But you are free to review my findings:

https://github.com/LazyTarget/ha-ica-todo

**Comment by mellamomax** [comment]
> I have successfully reverse engineered their API and authentication flow. I have not had the time to polish it or document it. But you are free to review my findings:
> 
> https://github.com/LazyTarget/ha-ica-todo

Interesting, will have a look! 

I haven’t looked into your integration yet but ICAs API isn’t sending out data, right? You have to manually retrieve it?

This is the repo I made - it works quite good for what the api can do. Or at least what I’ve managed to work out.

https://github.com/mellamomax/ica_shopping
